org.eclipse.scanning.malcolm.core/src/org/eclipse/scanning/malcolm/core/MalcolmDevice.java

```diff
@@ -192,19 +174,25 @@ public void eventPerformed(MalcolmEvent<Boolean> e) {
 	}
 
 	/**
-	 * Actions to take when the PointStart attribute is used
+	 * Called at the start of each point of the outer scan due to the {@link PointStart} attribute.
+	 * Gets the iterator of the points of the inner scan from the {@link SubscanModerator}.
 	 * @param moderator the SubscanModerator
 	 */
     @PointStart
     public void scanPoint(SubscanModerator moderator) {
-	Iterable<IPosition> scanPositions = moderator.getInnerIterable();
+    	final Iterable<IPosition> scanPositions = moderator.getInnerIterable();
         scanPositionIterator = scanPositions.iterator();
     }
 
-	protected void sendScanEvent(MalcolmEvent<MalcolmMessage> e) throws Exception {
-
-		MalcolmMessage msg      = e.getBean();
-		DeviceState newState = MalcolmUtil.getState(msg, false);
+	protected void sendScanEvent(MalcolmEvent<MalcolmMessage> event) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 20 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)